Update e2e test for domain connection (setup/domain flow)

gius80 opened the pull request:
Closes [DOMAINS-1886](https://linear.app/a8c/issue/DOMAINS-1886/create-an-e2e-test-for-the-happy-path)

## Proposed Changes
* Move `fillUseDomainIOwnInput`, `selectTransferYourDomain`, and `selectConnectYourDomain` methods from `DomainSearchComponent` to `UseADomainIOwnPage`
* Add container pattern support to `UseADomainIOwnPage` class (similar to `DomainSearchComponent`) to enable scoped element searches
* Update moved methods to use `getContainer()` for consistency and proper scoping

## Why are these changes being made?
This refactoring improves code organization by moving domain-specific methods to the appropriate page object class. The three methods (`fillUseDomainIOwnInput`, `selectTransferYourDomain`, and `selectConnectYourDomain`) are specifically related to the "Use a domain I own" flow and logically belong in `UseADomainIOwnPage` rather than the more generic `DomainSearchComponent`.

Additionally, adding container pattern support to `UseADomainIOwnPage` provides flexibility for future use cases where the component might appear within a specific container scope, and ensures consistency with the pattern already established in `DomainSearchComponent`.

## Testing Instructions

Run this e2e tests against wpcalypso:
```
CALYPSO_BASE_URL=https://wpcalypso.wordpress.com yarn workspace wp-e2e-tests test -- test/e2e/specs/flows/setup-domain__transfer-domain.ts
CALYPSO_BASE_URL=https://wpcalypso.wordpress.com yarn workspace wp-e2e-tests test -- test/e2e/specs/flows/setup-onboarding__use-a-domain-i-own-transfer-domain.ts
```

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

gius80 committed:
Update e2e test

gius80 committed:
Code refactoring

zaguiini reviewed (approved):
Changes look good. I ran the [Pre-Release E2E Tests](https://teamcity.a8c.com/buildConfiguration/calypso_calypso_WebApp_Calypso_E2E_Pre_Release/16314606) to make sure they'll pass in the test runners.

gius80 committed:
Update test to use UseADomainIOwnPage for domain input and transfer selection

- Replace DomainSearchComponent with UseADomainIOwnPage for domain input step
- Replace DomainSearchComponent with UseADomainIOwnPage for transfer selection step
- Add UseADomainIOwnPage import

gius80 committed:
Merge branch 'trunk' into add/DOMAINS-1886
